src/aurora/sacfile.cpp

```diff
@@ -0,0 +1,52 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Handling GFF3 files with SAC header.
+ */
+
+#include "src/common/encoding.h"
+#include "src/common/memreadstream.h"
+
+#include "sacfile.h"
+
+namespace Aurora {
+
+SACFile::SACFile(Common::SeekableReadStream &stream) : GFF3File(load(stream)) {
+	if (getType() != MKTAG('S', 'A', 'V', ' '))
+		throw Common::Exception("Invalid gff id");
+}
+
+Common::UString SACFile::getLevelFile() const {
+	return _levelFile;
+}
+
+Common::SeekableReadStream *SACFile::load(Common::SeekableReadStream &stream) {
+	stream.skip(4); // Unknown value, probably a version header?
+	uint32 nameLength = stream.readUint32LE(); // Length of the level identifier.
+
+	_levelFile = Common::readStringFixed(stream, Common::kEncodingASCII, nameLength); // The Level identifier string.
+
+	stream.skip(4); // Unknown value, probably a checksum?
+
+	return stream.readStream(stream.size() - stream.pos());
```

Comment:
Is there a reason you're doing the extra read/copy here?

You could instead construct a `Common::SeekableSubReadStream` for the range of the "inner" GFF data.

Of course, the parent stream needs to be for as long as the sub stream exists, so you need to change the `SACFile` constructor to take a `Common::SeekableReadStream *`, and take over that stream (and keep it around, and delete it when the `SACFile` is deconstructed, but, for example, stuffing it into a `ScopedPtr`).

This mirrors how `GFF3File` works already, too.